### PR TITLE
Add types to reporter

### PR DESCRIPTION
**Pull request by kasium**
Signed-off-by: Kai Mueller <15907922+kasium@users.noreply.github.com>

Part of #319

### CODE REVIEW

**Comment by kasium** [comment]
So hopefully this PR is better. I only added hints for public API methods/classes

**Review by yurishkuro — APPROVED** [review]

**Comment by kasium** [comment]
@yurishkuro not sure what to do with the failing coverage. I didn't change the logic, but I guess the number of lines make there a difference

File: =5 [hunk 1]
```diff
@@ -0,0 +1 @@
+Requirement already satisfied: tornado in ./venv/lib64/python3.7/site-packages (5.1.1)
```
Comment (yurishkuro) [inline =5:1]: ?

Comment (kasium) [reply =5]: No clue where this somes from

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**